src/ethereum_test_specs/blockchain.py

```diff
@@ -281,7 +281,7 @@ class BlockchainTest(BaseTest):
 
     pre: Alloc
     post: Alloc
-    blocks: List[Block]
+    blocks: List[Union[Block, Tuple[Block, Alloc]]]
```

Comment:
I would prefer to add a new field in the `Block` class and, if it's set, make the intermediate state verification.

Something like `Block.intermediate_post` or similar.

Reply:
Logically block doesn't have state.
Because one block can result in different satets depending on pre